staging/src/k8s.io/kubectl/pkg/polymorphichelpers/updatepodspec.go

```diff
@@ -85,6 +91,8 @@ func updatePodSpecForObject(obj runtime.Object, fn func(*v1.PodSpec) error) (boo
 		return true, fn(&t.Spec.JobTemplate.Spec.Template.Spec)
 
 	default:
-		return false, fmt.Errorf("the object is not a pod or does not have a pod template: %T", t)
+		return false, ResourceNotHavePodTemplate{
```

Comment:
Would it make sense to add a unit test case here (https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubectl/pkg/polymorphichelpers/updatepodspec_test.go) to check the type and the string? since the original test cases doesn't compare the error string